spec/components/schools/mentors/summary_component_spec.rb

```diff
@@ -0,0 +1,70 @@
+RSpec.describe Schools::Mentors::SummaryComponent, type: :component do
+  let(:school) { create(:school) }
+  let(:mentor_teacher) { create(:teacher, trs_first_name: 'Naruto', trs_last_name: 'Uzumaki') }
+  let(:started_on) { Date.new(2023, 9, 1) }
+  let(:mentor) { create(:mentor_at_school_period, teacher: mentor_teacher, school:, started_on:, finished_on: nil) }
+
+  context 'with no ECTs' do
+    it 'shows No ECTs assigned' do
+      render_inline(described_class.new(mentor:, school:))
+      expect(rendered_content).to have_css('.govuk-summary-list__row', text: 'Assigned ECTs')
+      expect(rendered_content).to have_css('.govuk-summary-list__value', text: 'No ECTs assigned')
+    end
+  end
+
+  context 'with less than or equal to 5 ECTs' do
+    let!(:ects) do
+      (1..5).map do |i|
+        teacher = create(:teacher, trs_first_name: "First name #{i}", trs_last_name: "Last name #{i}")
+        ect = create(:ect_at_school_period, teacher:, school:, started_on:, finished_on: nil)
+        create(:mentorship_period, mentor:, mentee: ect, started_on:, finished_on: nil)
+        teacher
+      end
+    end
+
+    it 'lists ECT names' do
+      render_inline(described_class.new(mentor:, school:))
+
+      ects.each do |teacher|
+        expect(rendered_content).to include("#{teacher.trs_first_name} #{teacher.trs_last_name}")
+      end
+    end
+  end
+
+  context 'with more than 5 ECTs' do
+    before do
+      (1..6).each do |i|
```

Comment:
If the index being zero based isn't an issue, could use `6.times`

Reply:
Also a nice improvement - thank you. 

Updated here: https://github.com/DFE-Digital/register-early-career-teachers-public/pull/476/commits/677e0a790f8cf4890233e55120a4d63c7cbeec98